src/components/ContactsList/Merging.vue

```diff
@@ -0,0 +1,629 @@
+						return list
+					}
+
+					// Init if needed
+					if (!list[property.name]) {
+						list[property.name] = []
+					}
+
+					list[property.name].push(property)
+					return list
+				}, {})
+		},
+
+		calculateConflictsToResolve() {
+			let conflictsCount = 0
+
+			this.usedProperties.forEach(property => {
+				if (this.conflictInformation[property]?.type === 'conflict' && this.resolvedConflicts.get(property) === undefined) {
+					conflictsCount++
+				}
+
+				if (this.conflictInformation[property]?.type === 'conflictWithMultipleValues' && !this.resolvedConflicts.get(property)?.size) {
+					conflictsCount++
+				}
+			})
+
+			if (this.chosenAddressBook === null) {
+				conflictsCount++
+			}
+
+			this.conflictsToResolve = conflictsCount
+		},
+
+		getPropertyValue(property) {
+			if (property.isMultiValue) {
+				// differences between values types :x;x;x;x;x and x,x,x,x,x
+				return property.isStructuredValue
+					? property.getValues()[0]
+					: property.getValues()
+			}
+			return property.getFirstValue()
+		},
+
+		sortUsedProperties() {
+			// the properties where this.conflictInformation[property].type === 'conflict' should have priority
+			return this.usedProperties.sort((a, b) => {
+				const aType = this.conflictInformation[a]?.type
+				const bType = this.conflictInformation[b]?.type
+
+				if (aType === 'conflict' && bType !== 'conflict') {
+					return -1
+				}
+				if (bType === 'conflict' && aType !== 'conflict') {
+					return 1
+				}
+				return 0
+			})
+		},
+
+		async mergeContacts() {
+			const contactToSave = this.contactsList[0]
+
+			const finalProperties = {}
+
+			this.usedProperties.forEach(property => {
+				if (this.conflictInformation[property]?.type === 'conflict') {
+					const resolvedVersion = this.resolvedConflicts.get(property)
+					if (resolvedVersion !== undefined) {
+						 finalProperties[property] = [this.dividedProperties[resolvedVersion][property]]
+					}
+				} else if (this.conflictInformation[property]?.type === 'conflictWithMultipleValues') {
+					const resolvedVersions = this.resolvedConflicts.get(property)
+					if (resolvedVersions?.size) {
+						finalProperties[property] = Array.from(resolvedVersions).map(version => this.dividedProperties[version][property])
+					}
+				} else if (this.conflictInformation[property]?.type === 'onlyInSecond') {
+					finalProperties[property] = [this.dividedProperties[1][property]]
+				} else {
+					finalProperties[property] = [this.dividedProperties[0][property]]
+				}
+			})
+
+			this.usedProperties.forEach(name => {
+				if (finalProperties[name] !== undefined) {
+					finalProperties[name].forEach((property) => {
+						contactToSave.vCard.updatePropertyWithValue(name, this.getPropertyValue(property))
+					})
+				}
+			})
+
+			contactToSave.groups = this.selectedGroups
+			contactToSave.addressbook = this.contactsList[this.chosenAddressBook.id].addressbook
+
+			await this.$store.dispatch('addContact', contactToSave)
```

Comment:
Can't figure out how to make this update on the server too, any help would be appreciated